src/lib/scheduler/abstract_scheduler.hpp

```diff
@@ -88,6 +84,9 @@ class AbstractScheduler : public Noncopyable {
   void schedule_and_wait_for_tasks(const std::vector<std::shared_ptr<AbstractTask>>& tasks);
 
  protected:
+  // Schedules the given tasks for execution and returns immediately.
+  static void _schedule_tasks(const std::vector<std::shared_ptr<AbstractTask>>& tasks);
+
   // Internal helper method that adds predecessor/successor relationships between tasks to limit the degree of
   // parallelism and reduce scheduling overhead.
   virtual void _group_tasks(const std::vector<std::shared_ptr<AbstractTask>>& tasks) const;
```

Comment:
Does it make sense to make this pure virtual?

Reply:
The immediate schedule does not have an implementation.

Reply:
My idea was to move the no-op implementation from `AbstractScheduler` to `ImmediateExecutionScheduler` and make the first purely virtual.

This is the implementaiton in `abstract_scheduler.cpp`:
https://github.com/hyrise/hyrise/blob/eab19a921b3066bb066d1fba124e682c2d4758e8/src/lib/scheduler/abstract_scheduler.cpp#L31-L33
This kinda calls for a purely virtual declaration. Also, as we won't end up with 20 different scheduler implementations, we won't have the case of 19 default (no-op) definitions an done specialization.

Reply:
Good point.